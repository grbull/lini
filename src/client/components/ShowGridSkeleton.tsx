import cn from 'classnames';
import React, { ReactElement } from 'react';

import { Skeleton } from './Skeleton';

export function ShowGridSkeleton(): ReactElement {
  return (
    <div className={cn('flex', 'flex-row', 'flex-wrap', 'px-2')}>
      {[1, 2, 3, 4, 5, 6, 7, 8, 9].map((i) => (
        <div className={cn('w-1/3', 'p-1')} key={i}>
          <Skeleton
            className={cn('w-full', 'h-0', 'relative', 'bg-gray-800')}
            style={{ paddingTop: `${(295 / 210) * 100}%` }}
          />
          <Skeleton className={cn('w-full', 'h-16', 'bg-gray-800', 'p-2')} />
        </div>
      ))}
    </div>
  );
}
