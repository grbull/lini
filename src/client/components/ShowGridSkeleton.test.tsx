/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect';

import { render, RenderResult } from '@testing-library/react';
import React from 'react';

import { ShowGridSkeleton } from './ShowGridSkeleton';

function setup(): RenderResult {
  const utils = render(<ShowGridSkeleton />);

  return { ...utils };
}

describe('ShowGridSkeleton Component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });
});
