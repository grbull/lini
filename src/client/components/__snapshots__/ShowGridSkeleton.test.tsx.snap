// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowGridSkeleton Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row flex-wrap px-2"
  >
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
    <div
      class="w-1/3 sm:w-1/4 md:w-1/5 p-1"
    >
      <div
        class="animate-pulse bg-gray-700 w-full h-0 relative bg-gray-800"
        style="padding-top: 140.47619047619045%;"
      />
      <div
        class="animate-pulse bg-gray-700 w-full h-16 bg-gray-800 p-2"
      />
    </div>
  </div>
</DocumentFragment>
`;
